raiden/network/proxies/user_deposit.py

```diff
@@ -360,6 +360,42 @@ def approve_and_deposit(
 
             self._deposit_check_result(transaction_sent, token, total_deposit, amount_to_deposit)
 
+    def plan_withdraw(self, amount: TokenAmount) -> BlockNumber:
+        """ Announce that you plan to withdraw tokens from the UserDeposit contract
+
+        Returns the block number at which the withdraw is ready.
+        """
+        estimated_transaction = self.client.estimate_gas(self.proxy, "planWithdraw", {}, amount)
+        if estimated_transaction is None:
+            failed_at_blocknumber = self.client.get_block(BLOCK_ID_LATEST)["number"]
+            raise RaidenRecoverableError(
+                f"planWithdraw gas estimation failed at {failed_at_blocknumber}"
+            )
+
+        transaction_sent = self.client.transact(estimated_transaction)
+
+        transaction_mined = self.client.poll_transaction(transaction_sent)
+        if not was_transaction_successfully_mined(transaction_mined):
+            raise RaidenRecoverableError("planWithdraw transaction failed")
+
+        return BlockNumber(transaction_mined.receipt["blockNumber"] + 100)
```

Comment:
Is `transaction_mined.receipt["blockNumber"]` correct? It looks like this gave me `9916158` instead of `9916165` for https://etherscan.io/tx/0x5c72180a27d112081e69e3917dc76746c6d5a6baa9399f963fede81f709fcf8e.

Reply:
Why does it look like you have the wrong number?

BTW, can't you rely on event polling to get this data?

Reply:
> Why does it look like you have the wrong number?

I used https://github.com/raiden-network/scenario-player/pull/544 to withdraw tokens and the resulting `ready_at_block` caused by that above transaction was `9916258`. There might be some other mistake, but I couldn't find any so far.

> BTW, can't you rely on event polling to get this data?

The event [does not include](https://etherscan.io/tx/0x5c72180a27d112081e69e3917dc76746c6d5a6baa9399f963fede81f709fcf8e#eventlog) the block number at which the withdraw is ready. It would make sense to add that, but for now we have to make do with the existing contract.